File: packages/urdu-js/tests/search/tokenize-search.test.ts
import { describe, it, expect } from 'vitest'
import { tokenizeForSearch } from '../../src/search/tokenize-search.js'

describe('tokenizeForSearch', () => {
  it('empty string returns empty array', () => {
    expect(tokenizeForSearch('')).toEqual([])
  })
  it('single word returns single token', () => {
    expect(tokenizeForSearch('علم')).toEqual(['علم'])
  })
  it('splits on spaces', () => {
    const tokens = tokenizeForSearch('علم اور ادب')
    expect(tokens).toHaveLength(3)
    expect(tokens).toContain('علم')
  })
  it('splits on Arabic comma U+060C', () => {
    const tokens = tokenizeForSearch('علم،ادب')
    expect(tokens).toHaveLength(2)
  })
  it('splits on Arabic semicolon U+061B', () => {
    const tokens = tokenizeForSearch('علم؛ادب')
    expect(tokens).toHaveLength(2)
  })
  it('strips diacritics from tokens', () => {
    const tokens = tokenizeForSearch('عِلمٌ')
    expect(tokens[0]).toBe('علم')
  })
  it('filters empty tokens after trimming', () => {
    const tokens = tokenizeForSearch('  علم  ')
    expect(tokens).toHaveLength(1)
    expect(tokens[0]).toBe('علم')
  })
  it('long sentence tokenizes all words', () => {
    const tokens = tokenizeForSearch('پاکستان کا قومی ترانہ بہت خوبصورت ہے')
    expect(tokens.length).toBeGreaterThan(4)
  })
})

File: packages/urdu-dart/test/search/tokenize_search_test.dart
import 'package:test/test.dart';
import 'package:urdu_tools/urdu_tools.dart';

void main() {
  group('tokenizeForSearch', () {
    test('empty string returns empty array', () {
      expect(tokenizeForSearch(''), equals([]));
    });
    test('single word returns single token', () {
      expect(tokenizeForSearch('رحم'), equals(['رحم']));
    });
    test('splits on spaces', () {
      final tokens = tokenizeForSearch('رحم اور آباد');
      expect(tokens, hasLength(3));
      expect(tokens, contains('رحم'));
    });
    test('splits on Arabic comma U+060C', () {
      final tokens = tokenizeForSearch('رحم،آباد');
      expect(tokens, hasLength(2));
    });
    test('splits on Arabic semicolon U+061B', () {
      final tokens = tokenizeForSearch('رحم؛آباد');
      expect(tokens, hasLength(2));
    });
    test('strips diacritics from tokens', () {
      final tokens = tokenizeForSearch('رُحَمً');
      expect(tokens[0], equals('رحم'));
    });
    test('filters empty tokens after trimming', () {
      final tokens = tokenizeForSearch('  رحم  ');
      expect(tokens, hasLength(1));
      expect(tokens[0], equals('رحم'));
    });
    test('long sentence tokenizes all words', () {
      final tokens = tokenizeForSearch('پاکستان کی قومی زبان اردو ہے');
      expect(tokens.length, greaterThan(4));
    });
  });
}
